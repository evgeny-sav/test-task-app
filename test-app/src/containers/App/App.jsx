import React from 'react';
import { shallowEqual, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { Container } from 'react-bootstrap';

import Loading from '../../components/Loading';

import './App.scss';
import Table from 'react-bootstrap/Table';

const userListToArray = list => {
  const usersArray = [];
  for (let index in list) {
    const user = list[index];
    usersArray.push(user);
  }

  return usersArray;
};

const App = () => {
  const history = useHistory();
  const { isLoading, userList } = useSelector(state => {
    return {
      isLoading: state.loading.isLoading,
      userList: state.userList,
    };
  }, shallowEqual);
  const handleUserClick = user => {
    history.push(`/users/${user.Id}`);
  };

  return isLoading ? (
    <Loading />
  ) : (
    <Container>
      <div className="App">
        <h1>App</h1>

        <Table striped bordered hover>
          <thead>
            <tr>
              <th>Id</th>
              <th>Firstname</th>
              <th>Surname</th>
            </tr>
          </thead>
          <tbody>
            {userListToArray(userList).map(user => (
              <tr
                key={user.Id}
                onClick={e => {
                  handleUserClick(user);
                }}
              >
                <td>{user.Id}</td>
                <td>{user.Firstname}</td>
                <td>{user.Surname}</td>
              </tr>
            ))}
          </tbody>
        </Table>
      </div>
    </Container>
  );
};

export default App;
