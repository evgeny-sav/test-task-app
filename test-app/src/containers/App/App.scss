.App {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: flex-start;
  justify-content: flex-start;

  min-height: calc(100vh - 200px);

  table {
    tr {
      cursor: pointer;
    }
  }
}
