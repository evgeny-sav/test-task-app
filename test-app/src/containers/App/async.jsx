import React, { lazy } from 'react';

export default lazy(() => import('./App'));
