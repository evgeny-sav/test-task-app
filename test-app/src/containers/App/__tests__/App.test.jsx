import React from 'react';
import { create } from 'react-test-renderer';
import Loading from '../../../components/Loading';
import App from '../App';

jest.mock('../../../components/Loading', () => 'Loading');
jest.mock('../../../components/Pagination', () => 'Pagination');
jest.mock('../../../components/PokemonCard', () => 'PokemonCard');

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: jest
    .fn()
    .mockImplementationOnce(() => {
      return {
        isLoading: true,
        pokemons: {
          nextLink: '',
          previousLink: '',
          count: 0,
          data: [],
        },
      };
    })
    .mockImplementationOnce(() => {
      return {
        isLoading: false,
        pokemons: {
          nextLink: 'next',
          previousLink: 'prev',
          count: 100,
          data: [{ name: 'name' }],
        },
      };
    }),
}));

describe('App', () => {
  it('should render loading', () => {
    const component = create(<App />);
    expect(component.root.findByType(Loading).type).toBe('Loading');
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with cards', () => {
    const component = create(<App />);
    expect(component.toJSON().children[0].props.className).toEqual('App');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
