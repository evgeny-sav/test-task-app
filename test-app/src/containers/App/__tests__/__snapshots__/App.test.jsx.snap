// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render loading 1`] = `<Loading />`;

exports[`App should render with cards 1`] = `
<div
  className="container"
>
  <div
    className="App"
  >
    <div
      className="pagination-btns"
    >
      <Pagination
        count={100}
        nextLink="next"
        page={1}
        previousLink="prev"
        setPage={[Function]}
      />
    </div>
    <div
      className="cards"
    >
      <PokemonCard
        pokemon={
          Object {
            "name": "name",
          }
        }
      />
    </div>
  </div>
</div>
`;
